src/Tuura/Plato/Translation.hs

```diff
@@ -1,17 +1,25 @@
 module Tuura.Plato.Translation where
 
 import Data.Char
-import Data.List.Extra
 
 import Tuura.Concept.Circuit.Basic
 import Tuura.Concept.Circuit.Derived
 
 data ValidationResult a = Valid | Invalid [ValidationError a] deriving Eq
 
+instance Monoid (ValidationResult a) where
+    mempty = mempty
+
+    mappend Valid (Invalid es) = Invalid es
+    mappend (Invalid es) Valid = Invalid es
+    mappend Valid Valid = Valid
+    mappend (Invalid es) (Invalid fs) = Invalid (fs ++ es)
+    -- mappend (Defined x) (Defined y) = if x == y then Defined x else Inconsistent
+
 data ValidationError a = UnusedSignal a
                        | InconsistentInitialState a
                        | UndefinedInitialState a
-                       | InvariantViolated a
+                       | InvariantViolated [Transition a]
```

Comment:
I'm actually having a big problem getting `InvariantViolated (Invariant a)` to work. 

In the function `ValidateFSM` in `FSM.Translation`, line 125 onwards, if I change this to use `InvariantViolated (Invariant a)`, it throws errors which state that `a` is actually a `Transition a`, and if I change it to use this, then it throws errors stating it expects type `a`, but it's actually `Transition a`. 

I'm quite stuck with this. I will do the other changes, and push, and we can discuss this.